src/js/profile/indicator.js

```diff
@@ -67,4 +73,29 @@ export class Indicator extends Component {
 
         }
     }
+
+    addIndicatorText(wrapper, title, indicatorData) {
+        let indicatorNode = indicatorClone.cloneNode(true);
+        $(indicatorTitleClass, indicatorNode).text(title);
+        $(chartDescClass, indicatorNode).text(indicatorData.description);
+        const isLink = !isNull(indicatorData.metadata.url) && !isEmptyString(indicatorData.metadata.url);
+
+        if (isLink) {
+            let ele = $('<a></a>');
```

Comment:
Could you please extract the metadata code into a separate method? That will help in future if we want to extends this class and only override the metadata.